animation_nodes/nodes/mesh/set_bevel_edge_weight.py

```diff
@@ -0,0 +1,31 @@
+import bpy
+from ... data_structures import VirtualDoubleList
+from ... base_types import AnimationNode, VectorizedSocket
+
+class BevelEdgeWeight(bpy.types.Node, AnimationNode):
+    bl_idname = "an_SetBevelEdgeWeight"
+    bl_label = "Set Bevel Edge Weight"
+    errorHandlingType = "EXCEPTION"
+
+    useWeightList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+        self.newInput(VectorizedSocket("Float", "useWeightList",
+            ("Weight", "weights"), ("Weights", "weights")))
```

Comment:
You don't have to use the same variable name, this doesn't really matter when using an execution function as opposed to execution code.

Reply:
Okay.
Omar, can you please tell me the nodes from which I can learn more about the execution function and execution code.

Reply:
https://github.com/3DSinghVFX/animation_nodes/blob/6f9437cab30c0158ef825116c13884352a6647c9/animation_nodes/nodes/material/cycles_material_output.py#L70
Another question is, the `required` we can only use when we use `yield` in a function?

Reply:
Animation Nodes converts a node tree into python code then executes it. Each node define the code that it emits/contributes in one of two ways:
- Execution function. In this case, the emitted code is a single function call to the execution function. This is the method you have been using so far. There isn't anything special about it.
- Execution code. In this case, the emitted code is what ever returned by the `getExecutionCode` function. We use this method in two cases:
   - We want to dynamically control the code that is emitted. For instance, if we have a node with three outputs, we don't need to compute the outputs that are not connected to anything. So we selectively emit the code that computes only the outputs that are linked. This is achieved using the `required` input, which tells us which outputs are required to be computed.
   - The execution code is very small. For instance, if your node adds two numbers, then you can just emit that code and avoid the extra overhead of a function call.

The `getExecutionCode` is a "generator". A generator behaves like an iterator. Every time you `yield` something, it is like you returned a new line of code. Eventually, all of those lines are combined, so it is not really special, just a convenient way to return multiple lines of code.

Execution code introduces some complications. To import a module, you have to implement the `getUsedModules` function so that AN imports it when preparing the execution unit. Also, if you want to use an AN function from the algorithms package—for instance—you have to use the full qualifying name (`AN.algorithms.[...]`).

Reply:
Omar, thank you so much for this information.